packages/flutter_tools/test/commands.shard/hermetic/build_macos_test.dart

```diff
@@ -175,14 +177,14 @@ STDERR STUFF
       const <String>['build', 'macos', '--no-pub']
     ), throwsA(isA<UsageException>()));
   }, overrides: <Type, Generator>{
-    Platform: () => notMacosPlatform,
-    FileSystem: () => fileSystem,
     ProcessManager: () => FakeProcessManager.any(),
     FeatureFlags: () => TestFeatureFlags(isMacOSEnabled: true),
   });
 
   testUsingContext('macOS build fails when feature is disabled', () async {
     final BuildCommand command = BuildCommand(
+      flutterUsage: TestUsage(),
+      platform: macosPlatform,
       androidSdk: FakeAndroidSdk(),
       buildSystem: TestBuildSystem.all(BuildResult(success: true)),
       fileSystem: MemoryFileSystem.test(),
```

Comment:
looks like we're mixing file systems here. Also, is anything grabbing the filesystem from the context override?

Reply:
We still need fileSystem override is used on `flutter_command.dart` on `verifyThenRunCommand`

Reply:
then you should pass the same one we're passing to the BuildCommand. Each test should have at most one MemoryFileSystem.